mp/src/game/server/momentum/tickset.cpp

```diff
@@ -7,13 +7,82 @@
 // memdbgon must be the last include file in a .cpp file!!!
 #include "tier0/memdbgon.h"
 
-float* TickSet::interval_per_tick = nullptr;
+static void OnIntervalPerTickChange(IConVar *var, const char *pOldValue, float fOldValue)
+{
+    ConVarRef tr(var);
+    float tickrate = tr.GetFloat();
+    if (CloseEnough(tickrate, TickSet::GetTickrate(), FLT_EPSILON))
+        return;
+
+    TickSet::SetTickrate(tickrate);
+}
+
+static void OnTickRateSet(const CCommand &command)
+{
+    // Search defined rates for one with a string matching the command argument.
+    for (unsigned rateIndx = TickSet::TICKRATE_FIRST; rateIndx < TickSet::TICKRATE_COUNT; rateIndx++)
+    {
+        if (!strcmp(command.ArgS(), TickSet::s_DefinedRates[rateIndx].sType))
+        {
+            TickSet::SetTickrate(TickSet::s_DefinedRates[rateIndx].fTickRate);
+            return;
+        }
+    }
+    Warning("Unknown tickrate. Use \"sv_interval_per_tick\" to set custom tickrates.");
+}
+
+static int OnTickRateAutocomplete(const char *partial,
+                                  char commands[COMMAND_COMPLETION_MAXITEMS][COMMAND_COMPLETION_ITEM_LENGTH])
+{
+    const int commandLength = Q_strlen("sv_tickrate ");
+
+    if (Q_strlen(partial) < commandLength)
+    {
+        // Checking for circumstances where partial is not as long as expected.
+        partial = "";
+    }
+    else
+    {
+        // Move start of the string to expected start of parameter.
+        partial += commandLength;
+    }
+
+    const unsigned partialLength = Q_strlen(partial);
+    int suggestionCount = 0;
+
+    // Search defined rates for one with a string matching the command argument.
+    for (unsigned rateIndx = TickSet::TICKRATE_FIRST; rateIndx < TickSet::TICKRATE_COUNT && suggestionCount < COMMAND_COMPLETION_MAXITEMS; rateIndx++)
+    {
+        if (!Q_strncmp(partial, TickSet::s_DefinedRates[rateIndx].sType, partialLength))
+        {
+            Q_snprintf(commands[suggestionCount], COMMAND_COMPLETION_ITEM_LENGTH, "sv_tickrate %s", TickSet::s_DefinedRates[rateIndx].sType);
+            ++suggestionCount;
+        }
+    }
+
+    return suggestionCount;
+}
+
+MAKE_CONVAR_C(sv_interval_per_tick, "0.015", FCVAR_MAPPING,
+    "Changes the interval per tick of the engine. Interval per tick is 1/tickrate, "
+    "so 100 tickrate = 0.01.", 0.001f, 0.1f, OnIntervalPerTickChange);
+
+static ConCommand sv_tickrate("sv_tickrate", OnTickRateSet,
+    "Changes the tickrate to one of a defined set of values. "
+    "Custom tickrates can be set using \"sv_interval_per_tick.\"",
```

Comment:
The strings of these ConVars aren't too long and can be recombined to one line

Reply:
Okay but it hurts D: